_integration-schemas/darksky/forecast.md

```diff
@@ -0,0 +1,292 @@
+---
+tap: "darksky"
+version: "1.0"
+
+name: "forecast"
+doc-link: "https://darksky.net/dev/docs#overview"
+singer-schema: "https://github.com/singer-io/tap-darksky/blob/master/tap_darksky/schemas/forecast.json"
+description: "This table contains information about the forcasts."
+
+replication-method: "Key-based Incremental"
+
+api-method:
+  name: "Time Machine Request"
+  doc-link: "https://darksky.net/dev/docs#time-machine-request"
+
+attributes:
+  - name: "forecast_date"
+    type: "date-time"
+    primary-key: true
+    description: ""
```

Comment:
This is an attribute the tap generates to extract data from the API, meaning there's no description of it in Darksky's API docs. There should still be one, though, since it determines how and what data is extracted during a replication job.

The [request parameters for the endpoint that the tap uses](https://darksky.net/dev/docs#time-machine-request) give us this clue:

> Either be a UNIX time (that is, seconds since midnight GMT on 1 Jan 1970) or a string formatted as follows: [YYYY]-[MM]-[DD]T[HH]:[MM]:[SS][timezone]. timezone should either be omitted (to refer to local time for the location being requested), Z (referring to GMT time), or +[HH][MM] or -[HH][MM] for an offset from GMT in hours and minutes. The timezone is only used for determining the time of the request; the response will always be relative to the local time zone.

And the endpoint is: `https://api.darksky.net/forecast/[key]/[latitude],[longitude],[time]`

Think of `[time]` as equivalent to `forecast_time`, as we have it in the table here.

TL:DR; This is the date that a forecast record pertains to for the given `latitude` and `longitude`.